### DISCUSSION TITLE
EF7 - Bulk updates and Bulk deletes

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#16] [General]
# EF7 - Bulk updates and Bulk deletes

Finally the long awaited bulk update and bulk delete feature is heading towards Entity Framework 7. So we will have a look how to use it and what might be some limitations.
Also I will have a quick look at the performance aspect, because this was the main driver for that feature.

https://steven-giesel.com/blogPost/09dd41f9-a0ca-497b-90db-19a89b68b8a0

**Comment by sam03041993** [comment]
A much awaited feature.. Thanks for this great article.

**Reply by linkdotnet** [reply]
Hey @sam03041993  thank you very much. 

And yes much awaited. They are really pushing some cool features out!

**Comment by AmrEbied** [comment]
Thanks for this great article, but if I need make an update to more items,What should I do?

**Reply by linkdotnet** [reply]
Thanks @AmrEbied. Can you explain a bit more your use case?
You can have multiple updates in one statement:
```csharp
ExecuteUpdateAsync(s => s
    .SetProperty(p => p.Age, p => 21)
    .SetProperty(...)
    .SetProperty(...));
```

**Reply by AmrEbied** [reply]
Great, thanks

**Comment by abhith** [comment]
{ thank you ♥ } for this very informative article on EF. 
Added to my [recommended developer stories](https://www.abhith.net/recommended/stories/) list 🚀.

**Reply by linkdotnet** [reply]
Thank you ver much! Cool page, keep it up!

**Comment by labeled** [comment]
Awesome work guys! Now to convince teams to drop all the third part extensions they so desperately cling to :D

**Reply by linkdotnet** [reply]
Exactly. The right step into the right direction. I hope some more use cases will follow.
